annotations/templates/annotations/repository_collection.html

```diff
@@ -127,7 +127,7 @@ <h5 class="list-group-item-heading mb-0">Group Texts</h5>
                                 <path d="M14 4.5V14a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V2a2 2 0 0 1 2-2h5.5zm-3 0A1.5 1.5 0 0 1 9.5 3V1H4a1 1 0 0 0-1 1v12a1 1 0 0 0 1 1h8a1 1 0 0 0 1-1V4.5z"/>
                               </svg>{% else %}{% endif %}</td>
                               <td>    
-                                <a href="{% if project_id %}{% url 'repository_text_import' repository.id group_id text.key project_id %}{% else %}{% url 'repository_text_import' repository.id group_id text.key %}{% endif %}" class="btn btn-primary btn-sm">Import</a>
+                                <a href="{% if project_id %}{% url 'repository_text_import' repository.id group_id text.key project_id %}{% else %}{% url 'repository_project_list' repository.id group_id text.key %}{% endif %}" class="btn btn-primary btn-sm">Import</a>
```

Comment:
looking at this more closely, just put the redirect url here directly instead of having the if statement in the view method.